x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java

```diff
@@ -484,14 +484,11 @@ public void testDisabledRealmsAreNotAdded() throws Exception {
     public void testAuthcAuthzDisabled() throws Exception {
         Settings settings = Settings.builder()
                 .put("path.home", createTempDir())
+                .put(XPackSettings.SECURITY_ENABLED.getKey(), false)
                 .put("xpack.security.authc.realms." + FileRealmSettings.TYPE + ".realm_1.order", 0)
                 .build();
         Environment env = TestEnvironment.newEnvironment(settings);
         Realms realms = new Realms(settings, env, factories, licenseState, threadContext, reservedRealm);
-
-        assertThat(realms.iterator().hasNext(), is(true));
```

Comment:
?

Reply:
When security is disabled `Realms#asList` returns an empty Collection